packages/create-workleap/prompts.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  isCancel,
+  cancel,
+  intro,
+  outro,
+  text,
+  confirm,
+  select,
+  multiselect,
+  note,
+} from "@clack/prompts";
+import pc from "picocolors";
+
+const DEFAULT_CANCEL_MESSAGE = "Operation cancelled.";
+
+type Primitive = Readonly<string | boolean | number>;
+export type Option<Value> = Value extends Primitive
+  ? {
+      value: Value;
+      label?: string;
+      hint?: string;
+    }
+  : {
+      value: Value;
+      label: string;
+      hint?: string;
+    };
+
+export const Intro = (title?: string) => {
+  if (title) {
+    intro(title);
+  }
+};
+
+export const Note = (message?: string): void => {
+  note(message);
+};
+
+export const Text = async (
+  message: string,
+  placeholder?: string,
+  defaultValue?: string,
+  customCancelMessage?: string
+): Promise<string> => {
+  const value = await text({
+    message,
+    placeholder,
+    defaultValue,
+  });
+
+  if (isCancel(value)) {
+    cancel(customCancelMessage ?? DEFAULT_CANCEL_MESSAGE);
+    process.exit(0);
+  }
+
+  return value;
+};
+
+export const Confirm = async (
```

Comment:
It's arguable if wrapping prompts primitives to hide the cancellation behavior is a good idea as it's less flexible in cases where [clack.group](https://github.com/natemoo-re/clack/tree/main/packages/prompts#grouping) is needed.

In those cases, the pattern would be broken or it would result in adding 1 to 1 function for every use case which result in a bad abstraction IMO.